### PR TITLE
[TRAFFIC-9315] - Update output fields for `confluent network describe`

### PR DESCRIPTION
**Pull request by syoung-cflt**
What
----

Note that this is merging into the feature branch https://github.com/confluentinc/cli/tree/traffic-8850-network-cli. We will merge commits into `main` from this feature branch after all of the commands for network are implemented

* Added new output fields for `confluent network describe` (discussed in previous PRs.)
* Updated golden files to reflect the changes
* Added new test cases for the conditional output
* Fixed `confluent network delete` after merging `main` into `traffic-8850-network-cli`. Updated it such that it supports deleting one or more networks.

References
----------

[CLI Commands Design](https://confluentinc.atlassian.net/wiki/spaces/PM/pages/2987394300/1-pager+--+Network+CLI)
[Implementation 1-Pager](https://confluentinc.atlassian.net/wiki/spaces/~712020463029eabbac401bbe1b033ce0d3a00e/pages/3136749631/Network+CLI)

Test & Review
-------------

Added integration tests.

Performed manual tests for all operations.

Open Questions / Follow-ups
---------------------------

### CODE REVIEW

File: internal/network/command.go [hunk 1]
```diff
@@ -15,29 +15,43 @@ import (
...
+	Zones                    []string `serialized:"zones"`
+	DnsResolution            string   `serialized:"dns_resolution"`
+	Phase                    string   `serialized:"phase"`
+	SupportedConnectionTypes []string `serialized:"supported_connection_types"`
+	ActiveConnectionTypes    []string `serialized:"active_connection_types"`
+	Vpc                      string   `serialized:"vpc"`
+	Account                  string   `serialized:"account"`
+	Project                  string   `serialized:"project"`
+	VpcNetwork               string   `serialized:"vpc_network"`
+	VNet                     string   `serialized:"vnet"`
```
Comment (brianstrauch) [inline internal/network/command.go:53]: What does "v" stand for?
```suggestion
	VNet                     string   `serialized:"v_net"`
```

File: internal/network/command.go [hunk 2]
```diff
@@ -77,44 +92,83 @@ func printTable(cmd *cobra.Command, network networkingv1.NetworkingV1Network) er
...
+		SupportedConnectionTypes: supportedConnectionTypes,
+		ActiveConnectionTypes:    activeConnectionTypes,
+	}
+
+	if phase == "READY" {
+		if network.Status.Cloud == nil {
+			return fmt.Errorf(errors.CorruptedNetworkResponseErrorMsg, "cloud")
+		}
+
+		if cloud == "AWS" {
```
Comment (brianstrauch) [inline internal/network/command.go:133]: `switch`?

File: internal/network/command.go (hunk 1 above)
Comment (syoung-cflt) [reply internal/network/command.go:53]: Azure's "Virtual Network", I am matching this with our API response here. Do you think that we should keep it consistent with the API response?
image

Comment (brianstrauch) [reply internal/network/command.go:53]: As someone who knows nothing about this networking feature, I think I'd prefer "Virtual Network" over "VNet"

File: internal/network/command.go (hunk 2 above)
Comment (syoung-cflt) [reply internal/network/command.go]: Updated!

File: internal/network/command.go (hunk 1 above)
Comment (syoung-cflt) [reply internal/network/command.go:53]: Just checked with our PM @kabraham13 that we think it's fair to assume users of this networking feature should know what a VPC or a VNet is, considering that users can specify which cloud provider to use while creating a Confluent network.

Can we keep it as "VNet"?

Comment (brianstrauch) [reply internal/network/command.go:53]: Sure, if it's a common term!

**Review by brianstrauch — APPROVED** [review]

**Review by ejagade — APPROVED** [review]

**Pull request merged**